lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -235,10 +235,16 @@ static void checkInheritanceClause(
       // AnyObject is not allowed except on protocols.
       if (layout.hasExplicitAnyObject &&
           !isa<ProtocolDecl>(decl)) {
-        decl->diagnose(canHaveSuperclass
+            if(!layout.isAnyObject()){
```

Comment:
I think you should add a comment here explaining why this must be a composition (since we know that `layout.hasExplicitAnyObject` is true).

Reply:
Added Comments